godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -517,3 +791,104 @@ macro_rules! impl_from_tuple_for_vector4x {
+           #[inline]
+           pub fn bezier_derivative(self, control_1: Self, control_2: Self, end: Self, t: real) -> Self {
+               Self::new(
+                   $(
+                       self.$comp.bezier_derivative(control_1.$comp, control_2.$comp, end.$comp, t)
+                   ),*
+               )
+           }
+
+            /// Returns the point at the given `t` on the [Bézier](https://en.wikipedia.org/wiki/B%C3%A9zier_curve)
+            /// curve defined by this vector and the given `control_1`, `control_2`, and `end` points.
+            #[inline]
+            pub fn bezier_interpolate(self, control_1: Self, control_2: Self, end: Self, t: real) -> Self {
+                Self::new(
+                        $(
+                            self.$comp.bezier_interpolate(control_1.$comp, control_2.$comp, end.$comp, t)
+                        ),*
+                )
+            }
+
+            /// Returns a new vector "bounced off" from a plane defined by the given normal.
+            #[inline]
+            pub fn bounce(self, normal: Self) -> Self {
+                -self.reflect(normal)
+            }
+
+            /// Returns the vector with a maximum length by limiting its length to `length`.
+            #[inline]
+            pub fn limit_length(self, length: Option<real>) -> Self {
+                let length = length.unwrap_or(1.0);
+
+                Self::from_glam(self.to_glam().clamp_length_max(length))
+
+            }
+
+            /// Returns a new vector moved toward `to` by the fixed `delta` amount. Will not go past the final value.
+            #[inline]
+            pub fn move_toward(self, to: Self, delta: real) -> Self {
+                Self::from_glam(self.to_glam().move_towards(to.to_glam(), delta))
+            }
+
+            /// Returns the result of projecting the vector onto the given vector `b`.
+            #[inline]
+            pub fn project(self, b: Self) -> Self {
+                Self::from_glam(self.to_glam().project_onto(b.to_glam()))
+            }
+
+            /// Returns the result of reflecting the vector defined by the given direction vector `n`.
+            #[inline]
+            pub fn reflect(self, n: Self) -> Self {
+                2.0 * n * self.dot(n) - self
+            }
+
+            /// Returns a new vector slid along a plane defined by the given normal.
+            #[inline]
+            pub fn slide(self, n: Self) -> Self {
+                self - n * self.dot(n)
+            }
+        }
+    };
+}
+
+/// Implements functions present on floating-point 3D and 4D vectors.
+macro_rules! impl_vector3_vector4_fns {
+    (
+        // Name of the vector type.
+        $Vector:ty,
+        // Names of the components, with parentheses, for example `(x, y, z, w)`.
+        ($($comp:ident),*)
+    ) => {
+        impl $Vector {
+            /// Returns the inverse of the vector.
```

Comment:
Doc comments like this one are not very useful, as it contains no extra information over the method name :wink: 
It should explain that the component-wise reciprocal is computed, or simply "1 / component".

Imo the method name is also really bad in Godot, too easy to confuse with matrix inverse or negated vector. Could we maybe call it [`recip` like in glam](https://docs.rs/glam/latest/glam/f32/struct.Vec3.html#method.recip)? + doc alias...